playground/cdk/CdkSample.AppHost/mykv.module.bicep

```diff
@@ -1,51 +1,32 @@
-targetScope = 'resourceGroup'
-
-@description('')
-param location string = resourceGroup().location
-
-@description('')
-param principalId string
-
-@description('')
-param principalType string
-
-@description('')
-param signaturesecret string
+resource mykv 'Microsoft.KeyVault/vaults@2019-09-01' = {
+    name: take('mykv-${uniqueString(resourceGroup().id)}', 24)
+    location: resourceGroup().location
+    properties: { }
+    tags: {
+        'aspire-resource-name': 'mykv'
+    }
+}
 
+output vaultUri string = mykv.properties.vaultUri
 
-resource keyVault_aMZbuK3Sy 'Microsoft.KeyVault/vaults@2022-07-01' = {
-  name: toLower(take('mykv${uniqueString(resourceGroup().id)}', 24))
-  location: location
-  tags: {
-    'aspire-resource-name': 'mykv'
-  }
-  properties: {
-    tenantId: tenant().tenantId
-    sku: {
-      family: 'A'
-      name: 'standard'
+resource KeyVaultAdministrator_mykv 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'KeyVaultAdministrator_mykv')
+    properties: {
+        principalId: principalId
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483')
+        principalType: principalType
     }
-    enableRbacAuthorization: true
-  }
+    scope: mykv
 }
 
-resource roleAssignment_hVU9zjQV1 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  scope: keyVault_aMZbuK3Sy
-  name: guid(keyVault_aMZbuK3Sy.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483'))
-  properties: {
-    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483')
-    principalId: principalId
-    principalType: principalType
-  }
+resource mysecret 'Microsoft.KeyVault/vaults/secrets@2019-09-01' = {
+    name: take('mysecret-${uniqueString(resourceGroup().id)}', 127)
+    properties: {
+        value: signaturesecret
+    }
+    parent: mykv
 }
 
-resource keyVaultSecret_mW5tlkNij 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
-  parent: keyVault_aMZbuK3Sy
-  name: 'mysecret'
-  location: location
-  properties: {
-    value: signaturesecret
-  }
-}
+param principalId string
 
-output vaultUri string = keyVault_aMZbuK3Sy.properties.vaultUri
+param principalType string
```

Comment:
I guess the order rendered is the order added to the construct. I think we should ask to have that fixed in the CDK itself. Parameters should be first.

Reply:
This is tied up with auto module splitting, but we eventually want it to be customizable.

Reply:
Moved params to the top and outputs to the bottom in https://github.com/Azure/azure-sdk-for-net/pull/45911